### ISSUE TITLE
Finding total volume for a cryptocurrency

### ISSUE THREAD

**Issue by everycoinprice** [ccxt/ccxt#15848]
- OS: MacOS
- Programming Language version: Node.js 18.12 LTS
- CCXT version: 2.2.35

So I am trying to calculate total volume of a cryptocurrency in the past 24 hours by adding all the baseVolume's from fetchTicker property of exchanges. Yet the total volume always comes up more than what popular ranking websites like coinmarketcap and coingecko post for the past 24 hours in volume

`

    const ccxt = require("ccxt");
    const binance = new ccxt.binance();
    const symbol = "BTC";
    let totalVolume = 0;
    binance.fetchTickers().then(function(tickers) {
          const tickValues = Object.values(tickers);
          for(let  t = 0; t < tickers.length; t++) {
              if(tickers[t].symbol.split("/")[0] == symbol) {
                  totalVolume  = totalVolume  + tickers[t].baseVolume;
             } // end of if
         } // end of for
       }); // end of then block
`

NOTE- This is just part of the code block (just for binance)- i repeat the same loop for all exchanges in ccxt.exchanges

Using this code i get totalVolume close to 80 million Bitcoins(BTC) traded in the past 24 hours, while coinmarketcap and coingecko report it close to 1 million

What i a missing?

Thanks for the help!

**Comment by samgermain** [comment]
could you share the coinmarketcap links that you're talking about? It's hard to know why it's different without anything to look into with regards to how they calculate total volume

I can point out that CCXT does not have support for every crypto exchange, so the actual number will be much higher than what you're calculating, I don't know how many exchanges coinmarketcap or coingecko list

**Comment by everycoinprice** [comment]
Coinmarketcap links-

Bitcoin's volume - https://coinmarketcap.com/currencies/bitcoin/
![Screenshot 2022-12-05 at 1 37 48 PM](https://user-images.githubusercontent.com/101936368/205585823-34614f76-dc2c-4297-b5b5-baea29f2f49a.png)


Volume methodology- https://support.coinmarketcap.com/hc/en-us/articles/360043395912-Volume-Market-Pair-Cryptoasset-Exchange-Aggregate-

**Comment by everycoinprice** [comment]
> could you share the coinmarketcap links that you're talking about? It's hard to know why it's different without anything to look into with regards to how they calculate total volume
> 
> I can point out that CCXT does not have support for every crypto exchange, so the actual number will be much higher than what you're calculating, I don't know how many exchanges coinmarketcap or coingecko list

Yes true, that is why CCXT's total volume for a cryptocurrency should be lower than that calculated by coinmarketcap or coingecko. As CCXT might be supporting lower exchanges than them. But, in reality it is higher. While both coingecko and coinmarketcap have similar volume numbers

For bitcoin's volume 24H on 05/12/2022->
Coinmarketcap says volume is [$21,443,030,192](https://coinmarketcap.com/currencies/bitcoin/markets/)
while Coingecko says [$22,598,703,601]https://www.coingecko.com/en/coins/bitcoin

while through CCXT would be at least 10 to 100 times higher than these

**Comment by ttodua** [comment]
The reason could be because ccxt includes not only spot markets, but some exchanges also load derivatives (swap, futures, ...) and thus, all tickers are including them too.   I don't remind whether CMC/CG might be including only spot, check that. If so, then you might try smth like this:

```
         if(tickers[t].symbol.split("/")[0] == symbol) {
              const market = binance.market(symbol);
              if (market.type === 'spot') {
                  totalVolume  = totalVolume  + tickers[t].baseVolume;
             }
```

**Comment by ttodua** [comment]
closing stale topic. let us know if anyone experiences any problems currently.
also note, CCXT (`fetchTickers` etc) get info from API, so whatever numbers exchanges provide us, it is what it is and we might not be able to equate it to CMC/CG calculations
